lib/matplotlib/rcsetup.py

```diff
@@ -310,6 +318,25 @@ def validate_colorlist(s):
         msg = "'s' must be of type [ string | list | tuple ]"
         raise ValueError(msg)
 
+def validate_cycler(s):
+    'return a cycler object from a string repr or the object itself'
+    if isinstance(s, six.string_types):
+        try:
+            # TODO: We might want to rethink this...
+            cycler_inst = eval(s, {'cycler': cycler, 'Cycler': Cycler})
```

Comment:
But we still would have to parse it.  By having users pass in the python
code we get to side step that.

This is also the repr of a Cycler so it is super easy for users to generate
this string.

On Sat, Aug 15, 2015 at 1:40 AM OceanWolf <notifications@github.com> wrote:

> In lib/matplotlib/rcsetup.py
> <https://github.com/matplotlib/matplotlib/pull/4686#discussion_r37133683>:
>
> > @@ -310,6 +318,25 @@ def validate_colorlist(s):
> >          msg = "'s' must be of type [ string | list | tuple ]"
> >          raise ValueError(msg)
> >
> > +def validate_cycler(s):
> > +    'return a cycler object from a string repr or the object itself'
> > +    if isinstance(s, six.string_types):
> > +        try:
> > +            # TODO: We might want to rethink this...
> > +            cycler_inst = eval(s, {'cycler': cycler, 'Cycler': Cycler})
>
> Ahh, of course, because the matplotlibrc passes in a string...
>
> Could we find a simpler way of passing in a cycle as a string? Asking the
> user to pass in code as a param feels bad. How about:
>
>
> (prop, list) op (prop, list) ...
>
> we could use a recursive function to deal with brackets, e.g. ((prop,
> list) + (prop, list)) * (prop, list)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4686/files#r37133683>.
>